wp-content/themes/goonj-crm/engine/helpers.php

```diff
@@ -24,12 +24,23 @@ function goonj_generate_activity_button( $activity, $office_id, $individual_id )
 	// Calculate the pending activities by comparing with $activityTypes
 	$pendingActivities = array_diff( $activityTypes, array( $completedActivityType ) );
 
+	// Set the 'You are Visiting as' field to 'Group' if both 'Entity_Type' and 'Entity_Name' 
+	// are provided in the Material Contribution form; otherwise, set it to 'Individual'.
+	$visitedAs = (!empty($activity['Material_Contribution.Entity_Type']) && !empty($activity['Material_Contribution.Entity_Name']))
+	? 'Group'
+	: 'Individual';
+
 	// Define the mapping for each activity to the corresponding button and redirect info
 	$activityMap = array(
 		'Office visit' => array(
 			'redirectPath' => '/processing-center/office-visit/details/',
-			'buttonText' => __( 'Proceed to Office Visit', 'goonj-crm' ),
+			'buttonText' => __( 'Proceed to Processing Center Tour', 'goonj-crm' ),
 			'queryParam' => 'Office_Visit.Goonj_Processing_Center',
+			'additionalParams' => array(
+				'Office_Visit.Entity_Type' => $activity['Material_Contribution.Entity_Type'], // Additional params from contribution activity
```

Comment:
`// Additional params from contribution activity`

Not need this comment as the code tells everything here